### ISSUE TITLE
Right Click on M1 Mac ??

### ISSUE THREAD

**Issue by GoldmanOz** [comfyanonymous/ComfyUI#904]
As the title suggests, I can't get the right click function to work on the Comfy grid, unable to change titles, colour of nodes etc..
M1 Macbook Pro, installed no problems, runs other than that.

**Comment by M1kep** [comment]
What browser are you using? Are you running the latest version of Comfy?
I personally can't reproduce this in Brave, Arc, or Firefox

**Comment by philholden** [comment]
Same issue M1 Mac. Ctrl+click is normally used for right click but this in not working in Comfy to not possible to get context menu.

**Comment by philholden** [comment]
Works with actual mouse. This suggest you are listening for e.which === 3 on click rather than listening for 'contextmenu' or click and crtl modifier.

**Comment by philholden** [comment]
@GoldmanOz very dirty but this worked for me. Paste this on to line 5886 of `web/lib/litegraph.core.js`. It will allow you to use Cmd+click as right mouse instead of Ctrl+click for now.

```js
if (e.metaKey) {
            
            //right button
                if (this.allow_interaction && !skip_action && !this.read_only){
                    
                    // is it hover a node ?
                    if (node){
                        if(Object.keys(this.selected_nodes).length
                           && (this.selected_nodes[node.id] || e.shiftKey || e.ctrlKey || e.metaKey)
                        ){
                            // is multiselected or using shift to include the now node
                            if (!this.selected_nodes[node.id]) this.selectNodes([node],true); // add this if not present
                        }else{
                            // update selection
                            this.selectNodes([node]);
                        }
                    }
                    
                    // show menu on this node
                    this.processContextMenu(node, e);
                }
                
            }
```

**Comment by GoldmanOz** [comment]
> @GoldmanOz very dirty temporary fix this worked for me. Paste this on to line 5886 of `web/lib/litegraph.core.js`. It will allow you to use Cmd+click as right mouse instead of Ctrl+click for now.
> 
> ```js
> if (e.metaKey) {
>             
>             //right button
>                 if (this.allow_interaction && !skip_action && !this.read_only){
>                     
>                     // is it hover a node ?
>                     if (node){
>                         if(Object.keys(this.selected_nodes).length
>                            && (this.selected_nodes[node.id] || e.shiftKey || e.ctrlKey || e.metaKey)
>                         ){
>                             // is multiselected or using shift to include the now node
>                             if (!this.selected_nodes[node.id]) this.selectNodes([node],true); // add this if not present
>                         }else{
>                             // update selection
>                             this.selectNodes([node]);
>                         }
>                     }
>                     
>                     // show menu on this node
>                     this.processContextMenu(node, e);
>                 }
>                 
>             }
> ```

Thank you for this, much appreciated....as much as I don't want to I've reverted to using a mouse to get by...Cheers

**Comment by elharo** [comment]
I can confirm this is still an issue. General UI principle for desktop apps is don't put any functionality only in the context menu. Many, perhaps most users, will never find it. There should be a way to get to this menu or these items without using the right mouse button.

Tangentially, don't intercept control-click on the Mac.

**Comment by elharo** [comment]
Yes, this is a reasonable workaround for the time being. Thanks for noting it.
